apps/angular/src/app/features/anime/table-view/table-view.component.ts

```diff
@@ -1,32 +1,145 @@
+const DEFAULT_SORT_DIRECTION = 'asc';
+const INITIAL_PAGE = 0;
+const INPUT_DEBOUNCE_TIME = 500;
+
 /** Table view component. */
 @Component({
   selector: 'app-table-view',
   templateUrl: './table-view.component.html',
   styleUrls: ['./table-view.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
 })
 export class TableViewComponent {
 
+  /** Total number of records for the current query. */
+  public animeListCount = 0;
+
+  /** Number of records per page. */
+  public pageSize = 5;
+
+  /** All possible type filters. */
+  public filterListByType: FilterItem[] = [];
+
+  /** Value of search input. */
+  public readonly search = new FormControl('');
+
+  /** Filter by type. */
+  public readonly typeFilter = new FormControl<Type[]>([Type.Tv]);
+
+  /** Current sort settings. */
+  public readonly sort$ = new BehaviorSubject<TableSort>({
+    field: DEFAULT_SORT_FIELD,
+    direction: DEFAULT_SORT_DIRECTION,
+  });
+
+  /** Current page number. */
+  public readonly currentPageNumber$ = new BehaviorSubject<number>(0);
+
   /** Table columns names. */
   public readonly displayedColumns: readonly string[] = ['image', 'title-english', 'title-japanese', 'aired-start', 'type', 'status'];
 
   /** Anime list. */
   public readonly animeList$: Observable<readonly AnimeBase[]>;
 
-  public constructor(private readonly animeService: AnimeService) {
-    this.animeList$ = this.getAnimeList();
-  }
+  public constructor(
+    urlService: UrlService,
+    private readonly animeService: AnimeService,
+  ) {
+    this.setFilterListByType();
+    this.setPageSize();
+    this.setInputValues();
 
-  private getAnimeList(): Observable<readonly AnimeBase[]> {
-    return this.animeService.fetchAnimeList().pipe(
-      map(pagination => pagination.results),
+    const paramsChange$ = this.search.valueChanges.pipe(
+      startWith(this.search.value),
+      combineLatestWith(
+        this.typeFilter.valueChanges.pipe(
+          startWith(this.typeFilter.value),
+        ),
+        this.sort$,
+      ),
+      tap(() => this.currentPageNumber$.next(INITIAL_PAGE)),
+      debounceTime(INPUT_DEBOUNCE_TIME),
     );
+
+    const params$ = paramsChange$.pipe(
+      combineLatestWith(this.currentPageNumber$),
+    );
+
+    this.animeList$ = params$.pipe(
+      switchMap(([[search, typeFilter, sort], pageNumber]) => {
+        const animeListOption = {
+          pageNumber,
+          sort,
+          filter: { byType: typeFilter !== null ? typeFilter : [Type.Tv] },
+          search: search !== null ? search : '',
+          limit: this.pageSize,
+        };
+        const animeListHttpParams = this.animeService.getAnimeListHttpParams(animeListOption);
+        urlService.setUrl(animeListHttpParams);
+        return this.animeService.fetchAnimeList(animeListHttpParams);
+      }),
+      map(animeList => {
+        this.animeListCount = animeList.count;
+        return animeList.results;
+      }),
```

Comment:
Ideally, map functions should be pure. And side effects should be in a `tap`. I suggest refactoring it like so:
```suggestion
      map(([[search, typeFilter, sort], pageNumber]) => {
        const animeListOption = {
          pageNumber,
          sort,
          filter: { byType: typeFilter !== null ? typeFilter : [Type.Tv] },
          search: search !== null ? search : '',
          limit: this.pageSize,
        };
        return this.animeService.getAnimeListHttpParams(animeListOption);
      }),
      tap(animeListHttpParams => urlService.setUrl(animeListHttpParams),
      switchMap(animeListHttpParams => this.animeService.fetchAnimeList(animeListHttpParams))
      tap(() => this.animeListCount = animeList.count)
      map(animeList => animeList.results),
```